server/planning/feed_parsers/events_ml.py

```diff
@@ -0,0 +1,171 @@
+
+from datetime import timedelta
+import logging
+
+from flask import current_app as app
+from xml.etree.ElementTree import Element
+
+from superdesk import get_resource_service
+from superdesk.io.feed_parsers import NewsMLTwoFeedParser
+from superdesk.metadata.item import (
+    ITEM_TYPE,
+    CONTENT_TYPE,
+    GUID_FIELD,
+    CONTENT_STATE,
+)
+from superdesk.errors import ParserError
+from superdesk.utc import utcnow
+
+logger = logging.getLogger(__name__)
+
+
+class EventsMLParser(NewsMLTwoFeedParser):
+    NAME = "eventsml"
+    label = "Events ML"
+
+    # map subject qcode prefix to scheme
+    # if value is None, no "scheme" is used, and name comes from qcode
+    # if value is not None, <name> element is used instead of qcode
+    SUBJ_QCODE_PREFIXES = {"subj": None}
+    missing_voc = None
+
+    def can_parse(self, xml: Element):
+        self.root = xml
+        try:
+            if not xml.tag.endswith("conceptItem"):
+                return False
+
+            concept = xml.find(self.qname("concept"))
+            if concept is None:
+                return False
+
+            type_node = concept.find(self.qname("type"))
+            if type_node is None:
+                return False
+
+            return type_node.attrib.get("qcode", "") == "cpnat:event"
+        except Exception:
+            return False
+
+    def set_missing_voc_policy(self):
+        # config is not accessible during __init__, so we check it here
+        if self.__class__.missing_voc is None:
+            self.__class__.missing_voc = app.config.get("QCODE_MISSING_VOC", "continue")
+            if self.__class__.missing_voc not in ("reject", "create", "continue"):
+                logger.warning(
+                    'Bad QCODE_MISSING_VOC value ({value}) using default ("continue")'.format(value=self.missing_voc)
+                )
+                self.__class__.missing_voc = "continue"
+
+    def parse(self, tree: Element, provider=None):
+        self.root = tree
+        self.set_missing_voc_policy()
+
+        try:
+            guid = tree.attrib["guid"]
+
+            if get_resource_service("events").find_one(req=None, guid=guid):
+                logger.warning("An event already exists with exact same ID. Updating events is not supported yet")
+                return []
+
+            item = {
+                GUID_FIELD: guid,
+                ITEM_TYPE: CONTENT_TYPE.EVENT,
+                "state": CONTENT_STATE.INGESTED,
+            }
+
+            self.set_occur_status(item)
+            self.parse_content_meta(tree, item)
+            self.parse_concept(tree, item)
+            self.parse_event_details(tree, item)
+
+            if item.get("firstcreated"):
+                item["event_created"] = item["firstcreated"]
+            if item.get("versioncreated"):
+                item["event_lastmodified"] = item["versioncreated"]
+
+            item["firstcreated"] = utcnow()
+            item["versioncreated"] = utcnow()
```

Comment:
not sure if we should override it, might be useful later when doing auto updating based on version